### PR TITLE
dotnet sdk update support api 2023

### PR DESCRIPTION
**Pull request by devinlstoen**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ProblemClassificationsClassificationResult.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            ServiceId = serviceId;
+            ProblemClassificationId = problemClassificationId;
+            ServiceIdRelatedServiceId = serviceIdRelatedServiceId;
+            DisplayName = displayName;
+            ResourceTypes = resourceTypes;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Identifier that may be used for solution discovery or some other purposes. </summary>
+        public string ProblemId { get; }
```
Comment (archerzz) [inline sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ProblemClassificationsClassificationResult.cs:79]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

File: sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ProblemClassificationsClassificationResult.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        /// <summary> Title of the problem classification result. </summary>
+        public string Title { get; }
+        /// <summary> Description of the problem classification result. </summary>
+        public string Description { get; }
+        /// <summary> Identifier of the service associated with this problem classification result. </summary>
+        public string ServiceId { get; }
```
Comment (archerzz) [inline sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ProblemClassificationsClassificationResult.cs:85]: same

**Review by archerzz — COMMENTED** [review]
Generally looks good to me. Need to confirm the format of some ID properties.

File: sdk/support/Azure.ResourceManager.Support/src/Generated/Models/LookUpResourceIdContent.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        internal LookUpResourceIdContent(string identifier, AzureSupportResourceType? azureSupportResourceType, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            Identifier = identifier;
+            AzureSupportResourceType = azureSupportResourceType;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> The System generated Id that is unique. Use supportTicketId property for Microsoft.Support/supportTickets resource type. </summary>
+        public string Identifier { get; set; }
```
[the remark this answers is not in the document]
Comment (devinlstoen) [reply sdk/support/Azure.ResourceManager.Support/src/Generated/Models/LookUpResourceIdContent.cs:65]: This is the support ticket id.

File: sdk/support/Azure.ResourceManager.Support/src/Generated/Models/LookUpResourceIdResult.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        /// <param name="resourceId"> The resource Id of support resource type. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        internal LookUpResourceIdResult(string resourceId, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            ResourceId = resourceId;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> The resource Id of support resource type. </summary>
+        public string ResourceId { get; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/support/Azure.ResourceManager.Support/src/Generated/Models/LookUpResourceIdResult.cs]: fixed, it's `arm-id`

File: sdk/support/Azure.ResourceManager.Support/src/Generated/Models/LookUpResourceIdContent.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/support/Azure.ResourceManager.Support/src/Generated/Models/LookUpResourceIdContent.cs:65]: string value, don't change the format

File: sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ClassificationService.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            ResourceTypes = resourceTypes;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Azure resource Id of the service. </summary>
+        public ResourceIdentifier ServiceId { get; }
+        /// <summary> Localized name of the azure service. </summary>
+        public string DisplayName { get; }
+        /// <summary> List of applicable ARM resource types for this service. </summary>
+        public IReadOnlyList<string> ResourceTypes { get; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ClassificationService.cs]: Fixed, it's `resource-type`

File: sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ProblemClassificationsClassificationResult.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ProblemClassificationsClassificationResult.cs]: String, don't need change the format

File: sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ProblemClassificationsClassificationResult.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/support/Azure.ResourceManager.Support/src/Generated/Models/ProblemClassificationsClassificationResult.cs]: Fixed, it's `arm-id`

**Review by archerzz — APPROVED** [review]

**Pull request merged**